aiida/schedulers/plugins/test_slurm.py

```diff
@@ -46,43 +54,46 @@ def test_parse_common_joblist_output(self):
 
-        job_held = 2
         job_held_parsed = len([j for j in job_list if j.job_state and j.job_state == JobState.QUEUED_HELD])
-        self.assertEquals(job_held, job_held_parsed)
+        self.assertEquals(JOBS_HELD, job_held_parsed)
 
-        job_queued = 2
         job_queued_parsed = len([j for j in job_list if j.job_state and j.job_state == JobState.QUEUED])
-        self.assertEquals(job_queued, job_queued_parsed)
+        self.assertEquals(JOBS_QUEUED, job_queued_parsed)
 
-        running_users = ['user5', 'user6']
         parsed_running_users = [j.job_owner for j in job_list if j.job_state and j.job_state == JobState.RUNNING]
-        self.assertEquals(set(running_users), set(parsed_running_users))
+        self.assertEquals(set(USERS_RUNNING), set(parsed_running_users))
 
-        running_jobs = ['862538', '861352', '863553']
         parsed_running_jobs = [j.job_id for j in job_list if j.job_state and j.job_state == JobState.RUNNING]
-        self.assertEquals(set(running_jobs), set(parsed_running_jobs))
-
-        self.assertEquals([j.requested_wallclock_time_seconds for j in job_list if j.job_id == '863553'][0], 30 * 60)
-        self.assertEquals([j.wallclock_time_seconds for j in job_list if j.job_id == '863553'][0], 29 * 60 + 29)
-        self.assertEquals([j.dispatch_time for j in job_list if j.job_id == '863553'][0],
-                          datetime.datetime(2013, 5, 23, 11, 44, 11))
-        self.assertEquals([j.submission_time for j in job_list if j.job_id == '863553'][0],
-                          datetime.datetime(2013, 5, 23, 10, 42, 11))
-        self.assertEquals([j.annotation for j in job_list if j.job_id == '863100'][0], 'Resources')
-        self.assertEquals([j.num_machines for j in job_list if j.job_id == '863100'][0], 32)
-        self.assertEquals([j.num_mpiprocs for j in job_list if j.job_id == '863100'][0], 1024)
-        self.assertEquals([j.queue_name for j in job_list if j.job_id == '863100'][0], 'normal')
-        self.assertEquals([j.title for j in job_list if j.job_id == '861352'][0], 'Pressure_PBEsol_0')
+        self.assertEquals(set(JOBS_RUNNING), set(parsed_running_jobs))
+
+        self.assertEquals(get_job('863553').requested_wallclock_time_seconds, 30 * 60)
+        self.assertEquals(get_job('863553').wallclock_time_seconds, 29 * 60 + 29)
+        self.assertEquals(get_job('863553').dispatch_time, datetime.datetime(2013, 5, 23, 11, 44, 11))
+        self.assertEquals(get_job('863553').submission_time, datetime.datetime(2013, 5, 23, 10, 42, 11))
+
+        self.assertEquals(get_job('863100').annotation, 'Resources')
+        self.assertEquals(get_job('863100').num_machines, 32)
+        self.assertEquals(get_job('863100').num_mpiprocs, 1024)
+        self.assertEquals(get_job('863100').queue_name, 'normal')
+
+        self.assertEquals(get_job('861352').title, 'Pressure_PBEsol_0')
+
+        self.assertEquals(get_job('863554').requested_wallclock_time_seconds, None)  # not set
```

Comment:
One thing I noticed is that this time will also be `None`, if the value could not be parsed.
Perhaps ok, but it means that finding a value `None` does no longer mean that parsing failed.